src/clr-angular/forms/datepicker/providers/datepicker-focus.service.ts

```diff
@@ -15,20 +15,23 @@ import { first } from 'rxjs/operators';
 export class DatepickerFocusService {
   constructor(private _ngZone: NgZone, @Inject(PLATFORM_ID) private platformId: Object) {}
 
-  // Credit: Material: https://github.com/angular/material2/blob/master/src/lib/datepicker/calendar.ts
   focusCell(elRef: ElementRef): void {
     this._ngZone.runOutsideAngular(() => {
-      this._ngZone.onStable
-        .asObservable()
-        .pipe(first())
-        .subscribe(() => {
-          if (isPlatformBrowser(this.platformId)) {
-            const focusEl = elRef.nativeElement.querySelector('[tabindex="0"]');
-            if (focusEl) {
-              focusEl.focus();
-            }
-          }
-        });
+      this.ngZoneIsStableInBrowser().subscribe(() => {
+        const focusEl = elRef.nativeElement.querySelector('[tabindex="0"]');
+        if (focusEl) {
+          focusEl.focus();
+        }
+      });
     });
   }
+
+  focusInput(element: HTMLInputElement): void {
+    this._ngZone.runOutsideAngular(() => this.ngZoneIsStableInBrowser().subscribe(() => element.focus()));
+  }
+
+  private ngZoneIsStableInBrowser() {
+    // Credit: Material: https://github.com/angular/material2/blob/master/src/lib/datepicker/calendar.ts
+    return this._ngZone.onStable.asObservable().pipe(first(), map(() => isPlatformBrowser(this.platformId)));
```

Comment:
I think you wanted to use `filter()`, not `map()`. With your code, you lost the `if (isPlatformBrowser(this.platformId))` from before, so you might crash server-side rendering.